iothub/service/src/Http/HttpClientFactory.cs

```diff
@@ -73,9 +72,9 @@ internal static HttpClient Create(string hostName, IotHubServiceClientOptions op
 
             httpMessageHandler.MaxConnectionsPerServer = DefaultMaxConnectionsPerServer;
             ServicePoint servicePoint = ServicePointManager.FindServicePoint(httpsEndpoint);
-            servicePoint.ConnectionLeaseTimeout = DefaultConnectionLeaseTimeout.Milliseconds;
+            servicePoint.ConnectionLeaseTimeout = s_defaultConnectionLeaseTimeout.Milliseconds;
```

Comment:
Tim, down here.

Reply:
Will this go away?

Reply:
No, this was the fix we provided so that a single http client instance would eventually start making requests to the correct endpoint if failover occurs.

Reply:
I thought we were going to let customers give us an HttpClient instance where this would already have been set. But I know we prefer to have smart defaults, so if we're capable of doing this, then I'm fine.

Reply:
Obviously, this one has nothing to do with the HttpClient instance, we just set it in the same method. I thought there was some problem netstandard2.0 where we couldn't set this value, though. Memory is hazy.

Reply:
> I thought we were going to let customers give us an HttpClient instance where this would already have been set. But I know we prefer to have smart defaults, so if we're capable of doing this, then I'm fine.

This setting fell under the category of a "smart default" last we talked about it. We agreed that our client's should be resilient to failover, and this same setting is applied to the default http client in track 2

> Obviously, this one has nothing to do with the HttpClient instance, we just set it in the same method. I thought there was some problem netstandard2.0 where we couldn't set this value, though. Memory is hazy.

The problem was that we had different HttpClientHandlers depending on the .NET target framework. The track 2 logic [here](https://github.com/Azure/azure-sdk-for-net/blob/cc96e605365ebfd30f9dd00a6dff47f89feb82d8/sdk/core/Azure.Core/src/Pipeline/ServicePointHelpers.cs#L87) demonstrates the problem.